parsons-tool-frontend/src/components/blocks/Block.js

```diff
@@ -74,7 +74,14 @@ const toFadedChildren = (text, fadedIndices, innerProps) => {
 export const defaultInnerProps = (currentInputs, index, setInput) => {
   const i = Math.floor(index / 2);
   return {
-    onChange: (e) => setInput(i, e.target.value),
+    onChange: (e) => {
+      setInput(i, e.target.value);
+      if (e.target.value.length > 10) {
+        e.target.style.width = `${e.target.value.length}ch`;
+      } else {
+        e.target.style.width = '10ch';
+      }
```

Comment:
I'd consider making 10 characters the minimum that it could be but this also good.

Reply:
wait could you clarify, with this condition 10ch is the minimum the width gets set to

Reply:
Ohh yes, I misunderstood the code - I'd have just done

```
e.target.style.width = `${Math.min(10, e.target.value.length)}ch`;
```

but it's the same thing as what you have anyway.

Reply:
rightio, I think I will keep the current version as its a bit more readable